### ISSUE TITLE
✨ Applications and Processes Widget

### ISSUE THREAD

**Issue by brindasasi** [GCTC-NTGC/gc-digital-talent#10788]
## ✨ Feature
 Job Applications is a component which loads and shows the user's PoolCandidate objects, divided in two accordions.

## 🕵️ Details

ALL applications, ie PoolCandidate objects, appear in the first accordion. Additionally, if a user was successfully qualified in a process, the PoolCandidate object _also_ appears in the the Recruitment processes section.

### Job Applications
Cards in the Job Applications accordion show the Job Title, a Status pill, classification, a date, and an icon.
The design is based on the same UI as the request card on the Manager dashboard. But please compare it to the design file again to make sure all the small details are correctly working.

The date is the finalDecisionDate, or the removed date, or the submittedDate, or the jobPoster closing date, in that order of priority.

**Q:** Should it be Removed date for removed applications?
**A:** Yes, if the candidate was removed, they should show up as "unsuccessful" with the "Assessed" date displaying the "Removed date" (but finalDecisionDate takes priority over removed)

The icon is a Edit icon for drafts and a magnifying glass for all others.

#### OUT OF SCOPE
Changes to Status Pills are **out of scope:** these will be updated in #12164. If this ticket is completed before that one, simply use the existing utility functions which generate status chips.

The dialogs which open are described in #12214.

### Recruitment Processes

If the user was Qualified in the process, it the PoolCandidate also appears in the Recruitement section. ie finalDecisionAt is not null and finalDecision is one of
  - QUALIFIED;
  - QUALIFIED_EXPIRED;
  - QUALIFIED_PLACED;
  - QUALIFIED_REMOVED;

Note, QUALIFIED_PENDING indicates they're likely to be qualified, but finalDecision has not _actually_ been made yet, and finalDecisionAt will be null. From the application's perspective, the application is still in assessment.

Cards in the Recruitment Processes accordion show the Job Title, a Status pill, classification, a number of skill assessments, a date, and an icon.
The date is a Removed date, if the user is removed, and Awarded date (finalDecisionDate) otherwise.
The icon is a magnifying class if the user for removed or expired applications, and an edit icon for others.

#### OUT OF SCOPE
Changes to Status Pills are **out of scope:** these will be updated in #12164. If this ticket is completed before that one, simply use the existing utility functions which generate status chips.

The dialogs which open are described in #12214.

## 🎨 Design File

https://www.figma.com/design/oZhQe9ahidHFfaiqgREvxa/Dashboard-(All-users)?node-id=2086-63995&node-type=frame&t=sltU21dpZZlO5BUt-0

### 🧑‍🎨 Designer

@JoshBeveridge 

## 📸 Screenshot
**Current dashboard**
![image](https://github.com/user-attachments/assets/11dd1b2c-ecd3-4e07-91b2-eba5d45708dc)

**New widget**
![image](https://github.com/user-attachments/assets/37910b1c-c3ae-4c8e-ac98-e40f3fba43fe)

## 🙋‍♀️ Proposed Implementation

The cards in the widget, and the widget component itself, should reuse the components built for the Manager dashboard (or be based on them).
- PreviewList component
- see PoolCandidateSearchRequestPreviewListItem and [ManagerDashboardPage](https://github.com/GCTC-NTGC/gc-digital-talent/blob/ce4c887b1491d9b1e30ab3d1c1ae5030bb20a69a/apps/web/src/pages/Manager/ManagerDashboardPage/ManagerDashboardPage.tsx) as examples

## 🌎 Localization

Statuses with longer descriptions...

## ✅ Acceptance Criteria
- [ ] Widget Title: Applications and processes
- [ ] Widget contains two sections/accordions, one for Job applications and one for Recruitment processes
- [ ] Job applications section also includes a link to Browse jobs page
- [x] Both accordions should use the PreviewList component, representing PoolCandidates as clickable cards which open 
- [x] Each application is a card in which the whole area is clickable and opens a dialog, as defined in #12214 
- [x] Cards in the Job Applications list:
  - [x] shows all PoolCandidates. Replaces the current "applications in progress" section.
  - [x]  show the Job Title, a Status pill, classification, a date, and an icon. 
  - [x] the date is the finalDecisionDate, or the removed date, or the submittedDate, or the jobPoster closing date.
  - [x] Edit icon for drafts and a magnifying glass for all others
- [x] The Recruitment processes list:
  - [x] Shows processes you've already qualified in. Replaces the "past applications" section. This _will_ repeat some items from the Job Applications section. To be specific, shows PoolCandidates where finalDecisionAt is not null and finalDecision is one of
    - QUALIFIED;
    - QUALIFIED_EXPIRED;
    - QUALIFIED_PLACED;
    - QUALIFIED_REMOVED;
  - [x] show the Job Title, a Status pill, classification, a number of skill assessments, a date, and an icon.
  - [x] The date is a Removed date, if the user is removed, and Awarded date (finalDecisionDate) otherwise.
  - [x] The icon is a magnifying class if the user for removed or expired applications, and an edit icon for others.
  - [ ]  Sort applications and processes in order of priority:
      - Drafts (closest deadline first)
      - Received
      - Under review
      - Application reviewed
      - Under assessment
      - Qualified in process
      - Unsuccessful
      - ~~Ineligible~~
      - Expired
    
## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] #11075 
- [ ] #12214
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/12236
```

Related, but not blocking
- [x] #12164
```[tasklist]
### Tasks
```

**Comment by brindasasi** [comment]
@substrae Its a bit weird to remove the view of statuses like  `Ready to hire`  from the dashboard main view to a different page under past applications. 🤔

**Comment by tristan-orourke** [comment]
Received, Under Review, Under Assessment Statuses:

Unlike for other statuses, there's no underReviewAt or underAssessmentAt for these statuses. In the current model, if an applicaiton has submittedAt, but no finalDecisionAt, then it's "Received", with no further distinctions.

But we now have business logic for these distinctions. When any interaction happens on the Screening and Assessment page, ie when an assessment has been created, the application moves into Under Review. Then, when it has passed step 1 of screening, it is under assessment. 

Should we add computed database fields for these to simplify work on the frontend? We're already doing some processing whenever assessments change.

**Comment by tristan-orourke** [comment]
I pulled the new status pills into its own ticket https://github.com/GCTC-NTGC/gc-digital-talent/issues/12164, but it shouldn't actually block this ticket. Since the current application/recruitment cards already compute their statuses on the current dashboard, we can use those for now.

**Comment by esizer** [comment]
> if a user was successfully qualified in a process, the PoolCandidate object also appears in the the Recruitment processes section.

It could be confusing for it to show in two different spots at the same time

**Comment by brindasasi** [comment]
@JoshBeveridge can the review application be read only pulling data from application snapshot to address the bug #12138

**Comment by petertgiles** [comment]
Let's request the design review in #10788 instead of stopping here
